src/compiler/checker.ts

```diff
@@ -36550,6 +36561,10 @@ namespace ts {
                 const meaning = name.parent.kind === SyntaxKind.TypeReference ? SymbolFlags.Type : SymbolFlags.Namespace;
                 return resolveEntityName(<EntityName>name, meaning, /*ignoreErrors*/ false, /*dontResolveAlias*/ true);
             }
+            else if (isJSDocEntryNameReference(name)) {

```

Comment:
This it the right place to support goto-def for the language service. But I don't *think* it adds support for renaming. Can you add a rename test? It might already work, but it might also require more changes in the binder or checker.